From 243d056ff1c2af583ceb67e5dfbfaac51dc96e63 Mon Sep 17 00:00:00 2001
From: Andi-Bogdan Postelnicu <abpostelnicu@me.com>
Date: Wed, 2 Jun 2021 14:08:11 +0300
Subject: [PATCH] Removed unused variable `supp_size` from
 plan_subset_encoding(...).

Builds with the latest layers fail as a result of stricter compiler
checks. This patch backports a fix upstream to remove an unused variable
which casues the build error. 

Upstream-Status: Backport [https://github.com/harfbuzz/harfbuzz/commit/243d056ff1c2af583ceb67e5dfbfaac51dc96e63]

Signed-off-by: Tony Tascioglu <tony.tascioglu@windriver.com>
---
 third_patry/harfbuzz-ng/src/src/hb-subset-cff1.cc | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/third_party/harfbuzz-ng/src/src/hb-subset-cff1.cc b/third_party/harfbuzz-ng/src/src/hb-subset-cff1.cc
index df322f845..35dae7b1f 100644
--- a/third_party/harfbuzz-ng/src/src/hb-subset-cff1.cc
+++ b/third_party/harfbuzz-ng/src/src/hb-subset-cff1.cc
@@ -402,7 +402,7 @@ struct cff_subset_plan {
   void plan_subset_encoding (const OT::cff1::accelerator_subset_t &acc, hb_subset_plan_t *plan)
   {
     const Encoding *encoding = acc.encoding;
-    unsigned int  size0, size1, supp_size;
+    unsigned int  size0, size1;
     hb_codepoint_t  code, last_code = CFF_UNDEF_CODE;
     hb_vector_t<hb_codepoint_t> supp_codes;
 
@@ -412,7 +412,6 @@ struct cff_subset_plan {
       return;
     }
 
-    supp_size = 0;
     supp_codes.init ();
 
     subset_enc_num_codes = plan->num_output_glyphs () - 1;
@@ -448,7 +447,6 @@ struct cff_subset_plan {
 	  code_pair_t pair = { supp_codes[i], sid };
 	  subset_enc_supp_codes.push (pair);
 	}
-	supp_size += SuppEncoding::static_size * supp_codes.length;
       }
     }
     supp_codes.fini ();
-- 
2.32.0

